### PR TITLE
chore(eks): update ALB controller versions

### PR DESCRIPTION
**Pull request by nmussy**
### Issue # (if applicable)

None as far as I could tell

### Reason for this change

Update the CDK listed ALB controller versions to match the current availability

### Description of changes

* Added missing controller versions
* Updated the Helm version of existing controller versions

### Description of how you validated changes

I listed the list of available versions by using the `ecr:ListImages` command on the `amazon/aws-load-balancer-controller` repository. I'm also filtering out tags that do not match a `v1.2.3` pattern, e.g. `v2.0.0-rc5`, `v2.0.0-test-linux_amd64`

For the Helm chart version, I **initially** manually went through the blame history of https://github.com/aws/eks-charts/blob/master/stable/aws-load-balancer-controller/Chart.yaml. @guessi then recommended I use the [Helm CLI](https://helm.sh/) to obtain the corresponding versions, which worked a ton better and was easily integrated to my tool:

```sh
helm repo add eks https://aws.github.io/eks-charts
helm repo update
helm search repo aws-load-balancer-controller --versions --output json
```

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by nmussy** [comment]
Updating the existing Helm chart versions changed the integ tests, not sure if I made a mistake by changing them. I can revert it or update the integs, let me know

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 1]
```diff
@@ -125,7 +130,7 @@ export class AlbControllerVersion {
   /**
    * v2.5.1
    */
-  public static readonly V2_5_1 = new AlbControllerVersion('v2.5.1', '1.5.2', false);
+  public static readonly V2_5_1 = new AlbControllerVersion('v2.5.1', '1.5.1', false);
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:133]: Not all chart version would aligned with App version, for the version `v2.5.1` the chart version should be `1.5.2`.

You may check version mapping with the command below
```sh
helm repo update
helm search repo aws-load-balancer-controller --versions
```

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 2]
```diff
@@ -90,7 +95,7 @@ export class AlbControllerVersion {
   /**
    * v2.4.2
    */
-  public static readonly V2_4_2 = new AlbControllerVersion('v2.4.2', '1.4.3', false);
+  public static readonly V2_4_2 = new AlbControllerVersion('v2.4.2', '1.4.2', false);
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:98]: Not all chart version would aligned with App version, for the version `v2.4.2` the chart version should be `1.4.3`.

You may check version mapping with the command below
```sh
helm repo update
helm search repo aws-load-balancer-controller --versions
```

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 3]
```diff
@@ -35,52 +35,57 @@ export class AlbControllerVersion {
...
    * v2.2.4
    */
-  public static readonly V2_2_4 = new AlbControllerVersion('v2.2.4', '1.4.1', false);
+  public static readonly V2_2_4 = new AlbControllerVersion('v2.2.4', '1.2.7', false);
 
   /**
    * v2.3.0
    */
-  public static readonly V2_3_0 = new AlbControllerVersion('v2.3.0', '1.4.1', false);
+  public static readonly V2_3_0 = new AlbControllerVersion('v2.3.0', '1.3.0', false);
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:78]: Not all chart version would aligned with App version, for the version `v2.4.0` the chart version should be `1.3.2`.

You may check version mapping with the command below
```sh
helm repo update
helm search repo aws-load-balancer-controller --versions
```

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 4]
```diff
@@ -35,52 +35,57 @@ export class AlbControllerVersion {
...
    * v2.1.2
    */
-  public static readonly V2_1_2 = new AlbControllerVersion('v2.1.2', '1.4.1', false);
+  public static readonly V2_1_2 = new AlbControllerVersion('v2.1.2', '1.1.4', false);
 
   /**
    * v2.1.3
    */
-  public static readonly V2_1_3 = new AlbControllerVersion('v2.1.3', '1.4.1', false);
+  public static readonly V2_1_3 = new AlbControllerVersion('v2.1.3', '1.1.5', false);
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:48]: `1.1.6`

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 5]
```diff
@@ -35,52 +35,57 @@ export class AlbControllerVersion {
   /**
    * v2.1.1
    */
-  public static readonly V2_1_1 = new AlbControllerVersion('v2.1.1', '1.4.1', false);
+  public static readonly V2_1_1 = new AlbControllerVersion('v2.1.1', '1.1.2', false);
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:38]: `1.1.3`

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 6]
```diff
@@ -20,12 +20,12 @@ export class AlbControllerVersion {
   /**
    * v2.0.0
    */
-  public static readonly V2_0_0 = new AlbControllerVersion('v2.0.0', '1.4.1', false);
+  public static readonly V2_0_0 = new AlbControllerVersion('v2.0.0', '1.0.6', false);
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:23]: `1.0.8`

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 7]
```diff
@@ -157,6 +162,16 @@ export class AlbControllerVersion {
...
+  /**
+   * v2.7.0
+   */
+  public static readonly V2_7_0 = new AlbControllerVersion('v2.7.0', '1.7.0', false);
+
+  /**
+   * v2.7.1
+   */
+  public static readonly V2_7_1 = new AlbControllerVersion('v2.7.1', '1.7.1', false);
+
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:174]: fyi, `2.7.2` just released today.

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts (hunk 1 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:133]: Thanks for the feedback, seems a lot more practical than using git tags :D I'll integrate that into my tool and update everything in a bit 👍

Comment (guessi) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:133]: Yup, sometimes there might have minor update for "chart" after "app" version released.

For `v2.5.1`, you may check it with the command below to ensure you are on the right track.
```sh
helm search repo aws-load-balancer-controller --versions --output json | jq '[.[] | select(.app_version == "v2.5.1")]'
```

Output:
```json
[
  {
    "name": "eks/aws-load-balancer-controller",
    "version": "1.5.2",
    "app_version": "v2.5.1",
    "description": "AWS Load Balancer Controller Helm chart for Kubernetes"
  },
  {
    "name": "eks/aws-load-balancer-controller",
    "version": "1.5.1",
    "app_version": "v2.5.1",
    "description": "AWS Load Balancer Controller Helm chart for Kubernetes"
  }
]
```

Comment (guessi) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:133]: There's more about [helm search repo](https://helm.sh/docs/helm/helm_search_repo/), check it out!

Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:133]: Yeah that's the command I'm going with, I already have all the SDK versions fetched so I'll just do the one request in JSON output and find each corresponding helm version, shouldn't take long, I just won't be able to work on it right away. Thanks again 👍

**Comment by nmussy** [comment]
`AlbControllerVersion` should be up to date now, I'll update the integs in a bit 👍

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts (hunk 7 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:174]: Added

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts (hunk 5 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts]: Fixed 👍

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts (hunk 4 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts]: Fixed 👍

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts (hunk 3 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts]: Fixed 👍

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts (hunk 2 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts]: Fixed 👍

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts (hunk 1 above)
Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts]: Fixed 👍

File: packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts [hunk 8]
```diff
@@ -157,6 +162,21 @@ export class AlbControllerVersion {
...
+  /**
+   * v2.7.1
```
Comment (guessi) [inline packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts:176]: Comment here should be v2.7.2

Comment (nmussy) [reply packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts]: Fixed, thanks

**Review by paulhcsun — APPROVED** [review]
Thanks for updating these @nmussy! And good callout on using the Helm CLI @guessi!

**Comment by paulhcsun** [comment]
@mergifyio update

**Comment by paulhcsun** [comment]
Hey @nmussy, after discussing with the team we've decided to revert this PR to err on the side of caution in case there is any unintended consequences with resources being replaced. Before I'd like to accept the change again I just have some questions to clarify:

1. Were there any integ test changes? 
2. Could you explain what you meant by the tests no longer being necessary with the helm CLI output, and that the Git tags were incorrect?
3. Are you able to add any tests to verify that this would not be a breaking change that would not be a breaking change?
4. Is there any other reason for this change besides matching the versions to the current availability? It seems like it was working fine before as there were no reported issues.

**Comment by nmussy** [comment]
Hey @paulhcsun, sorry for the confusion.
1. No, there were integ tests change initially, but using the helm CLI resolved this issue
2. Before @guessi mentioned the helm CLI, I used the git file history of https://github.com/aws/eks-charts/blob/master/stable/aws-load-balancer-controller/Chart.yaml, as indicated in the comments of the class:
https://github.com/aws/aws-cdk/blob/ec94f1a03dcf46960a32df5861ccca0cccbfcdc5/packages/aws-cdk-lib/aws-eks/lib/alb-controller.ts#L13-L18
This turned out to be erroneous, and I'd updated more versions than I should have, see ec94f1a03dcf46960a32df5861ccca0cccbfcdc5. This caused integ tests changes. After switching to the CLI, only the controller versions with the `1.4.1` version required a change. These controller versions were introduced by #25330, by @colifran
3. I'm not really sure how I would test that. It would definitely cause a change, but I don't know if it would be a breaking one
4. No reason other than matching the Helm version

If we want to err on the side of caution I can reopen this PR and only add the new versions, let me know 👍

**Comment by paulhcsun** [comment]
No worries at all! I jumped the gun on this one. I think let's just stick with adding the new versions for now until we have a better way of testing it helm chart version changes for the existing versions. The fact that no snapshots are changing could be due to a testing gap but I will confirm that with @colifran and get back to you. 

For newly added versions it looks like we can use them in the existing unit and integ tests per https://github.com/aws/aws-cdk/pull/27910. It looks like we'd also need to add in the `policy.json` files for each of the new versions we're adding. I'm also checking with a team member to confirm if there are any other ways to test these changes.

**Comment by nmussy** [comment]
Hey @paulhcsun, I just opened #29959, we can move our conversation there. Thanks for the help 👍

**Pull request merged**